### DISCUSSION TITLE
How can I create field based permission for list/detail queries?

### DISCUSSION THREAD

**Discussion by michaldev** [strapi/strapi#8079] [Legacy Topics]
I would like to create restrictions for list/get operations. 

Use case 1: I would like to show items, when "user" field is assigned for me.
Use case 2: I would like to show items, when "user" field is not assigned for me.

Filters are not helpful, because I would like block other queries.

**Comment by lauriejim** [comment]
Hello!
You can use the same system as in this video - https://www.youtube.com/watch?v=RBfPsmupU70&list=PL7Q0DQYATmvhlHxHqfKHsr-zFls2mIVTi&index=18
You are forcing to filter on a specific attribute.
For your use case, you will have to use `ctx.state.user.id` to get the user ID of the user that execute the request (authenticated request) and then force the filter on your related field.
